core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -166,21 +167,65 @@ private[history] class FsHistoryProvider(conf: SparkConf) extends ApplicationHis
             newLastModifiedTime = math.max(newLastModifiedTime, modTime)
             modTime > lastModifiedTime
           } else {
-            false
+            val appLogStatus = fs.listStatus(new Path(dir.getPath().toUri()))
+            val appAttemptsDirs = if (appLogStatus != null) appLogStatus.filter(_.isDir).toSeq 
+                                  else Seq[FileStatus]()
+            var isValidApplicationLogToUpdate = false
+            breakable {
+                for (appAttemptDir <- appAttemptsDirs){
+                    // There are multiple attempts inside this application logs
+                    if (fs.isFile(new Path(appAttemptDir.getPath(),
+                                           EventLoggingListener.APPLICATION_COMPLETE))) {
+                      val modTime = getModificationTime(dir)
+                      newLastModifiedTime = math.max(newLastModifiedTime, modTime)
+                      isValidApplicationLogToUpdate = modTime > lastModifiedTime
+                      break
+                    }
+                }
+            }
```

Comment:
please do not use break here. While it's unnatural that Scala does not provide native support for breaks, we try to avoid breaks elsewhere in the code base because it requires us to wrap the code in a `breakable` block. Here you can just use a while loop with a boolean var instead.